### PR TITLE
Baseline Azure code for Fabric deploy

### PR DESCRIPTION
**Pull request by krisvishwanathan**
Hey Tong,

Just as we discussed here is the code for deploying Fabric on Azure code. We tested the green path provisioning and destroying all the required resources including Fabric. This is the first stab for end-to-end Fabric deployment and delete in Microsoft Azure, stay tuned for more tweaks and cleanup. 

To test this use below steps:
1. Goto Azure portal and create a user from Active Directory portal menu option
2.  You will have to update the  vars/azure.yml with you ad_user and subscription info)
3. To deploy azure VMs and fabric
ansible-playbook -e "mode=apply password=xxx" deploy_fabric_azure.yml -vvv
4. To delete azure VMs and fabric
ansible-playbook -e "mode=destroy password=xxx" deploy_fabric_azure.yml -vvv
 
I will be updating the readme file soon for detailed instructions.

### CODE REVIEW

File: vars/azure.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+# Azure  Active Directory or Service Prinicpal Keys will be used to provision VMs  on Azure Cloud
+auth: {
+  
+  subscription_id: 5b713e38-dec5-4a1f-bb95-5303b9c6d7ec,
```
Comment (litong01) [inline vars/azure.yml]: how to get this thing? Can it be retrieved?

File: vars/azure.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  azure_secret_key: "{{ secret | default(lookup('env', 'AZURE_SECRET')) }}", 
+  azure_tenant: " "
+
+}
+
+# These variable defines Azure cloud provision attributes
+cluster: {
+  region_name: "eastus",     #TODO  Dynamic fetch
+  resource_group: "fabric",
+  vm_size: "Standard_D1",
```
Comment (litong01) [inline vars/azure.yml]: This variable should be removed. Use the flavor_name variable.

File: vars/azure.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+  target_os: "UbuntuServer",
+  image_name: "ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-*",
```
Comment (litong01) [inline vars/azure.yml]: are these two variables still needed? If not, remove them.

File: vars/azure.yml (hunk 1 above)
Comment (krisvishwanathan) [inline vars/azure.yml]: Yes, I include the steps in the readme file.

File: vars/azure.yml (hunk 2 above)
Comment (krisvishwanathan) [inline vars/azure.yml]: Are you asking to remove cluster variable and instead use the variable name as flavor_name?

**Pull request merged**